### PR TITLE
Warn on breaking function signatures

### PR DESCRIPTION
**Pull request by iwahbe**
Warn on breaking changes caused by https://github.com/pulumi/pulumi/issues/13563.

### CODE REVIEW

**Review by danielrbradley — APPROVED** [review]
The switch with tuple-like struct looks a bit weird in Go but looks functional.

File: internal/cmd/compare.go [hunk 1]
```diff
@@ -43,22 +46,33 @@ func compareCmd() *cobra.Command {
...
+	var schOld schema.PackageSpec
+	schOldDone := make(chan error)
+	go func() {
+		var err error
+		schOld, err = pkg.DownloadSchema(ctx, repository, provider, oldCommit)
+		if err != nil {
+			cancel()
+		}
+		schOldDone <- err
+	}()
```
[the remark this answers is not in the document]
Comment (iwahbe) [reply internal/cmd/compare.go:67]: I'm not aware of any language tools to make testing this easer. We could add a flag to `pkg.DownloadSchema` to allow us to test this, but it doesn't have language support.

[the remark this answers is not in the document]
Comment (danielrbradley) [reply internal/cmd/compare.go:67]: Guessing you just manually tested this locally then?

[the remark this answers is not in the document]
Comment (iwahbe) [reply internal/cmd/compare.go:67]: Yep

**Pull request merged**